### ISSUE TITLE
Arg-completer respect namespaces?

### ISSUE THREAD

**Issue by carson-r** [elves/elvish#1267]
## tl;dr:
using an alias, importing `foo:bar` as `f:bar` via `use foo f` will result in `edit:completion:arg-completer[foo:bar]` not running, and instead calling `edit:completion:arg-completer[f:bar]`.

I'm writing a elvish library for xbps, and have a file, `~/.elvish/lib/xbps.elv`. At the top of my `rc.elv`, I call `use xbps x`. This is convenient, as all the commands are now very quick to type. However, in `xbps.elv`, I want to define an arg-completer. It seems that for a function foo, to work in my shell, I have to write `edit:completion:arg-completer[x:foo]`. `edit:completion:arg-completer[xbps:foo]` and `edit:completion:arg-completer[foo]` do not work. The arg-completer code is in `xbps.elv`.

Is there a way to write argument completion that survives a import alias?

**Comment by krader1961** [comment]
I can't speak to whether the arg-completer should automatically recognize `x:foo` as an alias for `xbps:foo` but note that for interactive use you can use the [recently introduced](https://elv.sh/blog/0.15.0-release-notes.html) `edit:add-var` to define `foo` as an interactive alias for `xbps:foo`. So one solution is to use `foo` in the arg-completer definition and `edit:add-var foo~ $xbps:foo~` to put the command in the interactive namespace.

**Comment by zzamboni** [comment]
It would be interesting if there was a way, within the module, to know the name/alias with which it is being loaded. Then the completer definitions could be set automatically using the correct names.

As a workaround, you can do the mapping by hand after loading the module. For example, for the `epm:` completions:
```
[~]|> use epm e
[~]|> keys $edit:completion:arg-completer | each [k]{
        if (str:has-prefix $k epm:) {
          edit:completion:arg-completer[(str:replace epm: e: $k)] = $edit:completion:arg-completer[$k]
        }
      }
```

**Comment by carson-r** [comment]
@krader1961 's solution is a good one, I was unaware of the update. Closing this issue.
